code/stable/projectile/projectile_u_p_l_b_wi_v_f/src/swift/README.md

```diff
@@ -1,7 +1,13 @@
 # Projectile 
 > Authors:  Samuel J. Crawford, Brooks MacLachlan, W. Spencer Smith
 
-> Purpose: efficiently and correctly predict whether a launched projectile hits its target.
+> Purpose: efficiently and correctly predict whether a launched projectile hits its target
+
+> Motivation: Projectile motion is a common problem in physics.
+
+------------------------------------------------------------
+## What 
+ Common examples of projectile motion include ballistics problems (missiles and bullets) and the flight of the balls in various sports (baseball, golf, football, etc.).
```

Comment:
When I think of "motivation" related to software, I _think_ that I think (lol) strictly technical features:
1. What makes this software special?
2. What differentiates this software from other, similar, software?

However, this (the linked code) answers the question of "what is the motivation of the problem this software somehow solves?," which I think is more intended for a non-technical audience. So, I'm not quite sure who our audience is for this information (technical/non-technical/mix?).

If a software (assuming it were publicly available) is large/prominent enough such that it would have this kind of information, I imagine it would have a website, where this kind of marketing material would be found instead. Conversely, if the software isn't, then I imagine it would be somewhat self-evident to the audience of developers seeking out the software (they probably have a specific issue in mind, searching for a few specific keywords, and then looking at program/library documentation).

Reply:
Addendum to this after my latest comment below: the answers to (1) and (2) in a motivation section would be terse (e.g., X was too slow / A goes fast, Y is purely CLI-based / A creates a pretty GUI, Z is... / A has..., etc.), so it would not be as complete as the potential "Software Review"/Comparison Chart.

Reply:
These could potentially be even more terse; 
> A solves various problems with the following alternatives: X is too slow, Y is purely CLI-based (while A creates a pretty GUI), .…

By formatting it this way, we have the flexibility to comment on A's functionality/qualities when necessary for completeness, but don't need to when it is redundant (although this completeness might be desired); a table may also be a nice way of formatting this data.

Reply:
Yep, that's better. Though, I think the more complete, less marketing-like version, is reserved for the "Software Review" / Comparison Chart, but I agree with you. I think a good example of what we're talking about here is the Gitea project, where, in their [README.md](https://github.com/go-gitea/gitea), they give very quick highlights in the Purpose, and then have a dedicated page for [software comparison](https://docs.gitea.com/installation/comparison).

Reply:
If the bar for the motivation is to do something better than the competition, than none of our case study projects have a motivation.  :smile:  Research and commercial projects should have the kind of motivation that distinguished the project from the "competition."  Although our software doesn't have this kind of motivation, since it doesn't do anything better than the competition, I still think we can motivate it.  Of course we have the motivation for our Drasil research project, but that isn't the motivation I'm thinking.  The motivation is for why have software to solve the kind of problems the software solves.  It isn't motivation for why the current software is the best, but rather the motivation for having software to solve the problem.  Projectile motion is an important physics problem.  A software tool to help with projectile motion will help physicists/engineers/scientists.

If we had enough examples, either real or brainstormed, we could identify the patterns in the Motivation statements and build a language for generating motivation statements.   For now, I think I'm fine with a non-parameterized statement that we make for each example.

Reply:
You've convinced me :smile: 

> A software tool to help with projectile motion will help physicists/engineers/scientists.

Do you think we can replace the existing "What" section with your quote (or a variation of it)?

Reply:
Yes, this does sound like a better "What".  I do think we also need examples, but probably not in the README file.  I think of the examples as more something that goes in the SRS to help define the scope.  In projectile motion if the scope is for velocities less than those for ballistics, we can use the example of the speed of the balls in various sports.  The example and the scope are related.  They are also related to assumptions and typical values.  I don't think this information belongs in the README; I just mention it as I sort out my thoughts on the matter.  :smile: